### PR TITLE
[PSM Interop] New cleanup script

### PR DESCRIPTION
**Pull request by sergiitk**
1. Changes the resource retention period to 2 days for all resources (previously 7 days for TD resources, 6 hours for k8s). This solved a problem with k8s resources being stuck because corresponding TD resources weren't deleted.
2. Resume on namespace cleanup failures
3. Add secondary lb cluster cleanup logic
4. Modularize `grpc_xds_resource_cleanup.sh`
5. Make `KubernetesNamespace`'s methods `pretty_format_status` and `pretty_format_metadata` public
6. `pretty_format_status`: also print resource kind, creation and deletion requested dates

ref b/259724370, cl/517235715

### CODE REVIEW

**Comment by sergiitk** [comment]
Test run attempts:
- [ ] [grpc/core/master/linux/grpc_xds_resource_cleanup](https://source.cloud.google.com/results/invocations/90e2c8f5-b4aa-4c8a-8cc3-9f830bae2592)

**Comment by sergiitk** [comment]
Latest test run:
- [ ] [grpc/core/master/linux/grpc_xds_resource_cleanup](https://source.cloud.google.com/results/invocations/8178f78e-fcfc-41cc-9c4c-8053fc63033e)

File: tools/internal_ci/linux/grpc_xds_resource_cleanup.sh [hunk 1]
```diff
@@ -13,61 +13,124 @@
...
+cleanup::job::cleanup_cluster_security() {
+  cleanup::activate_cluster GKE_CLUSTER_PSM_SECURITY
+  cleanup::run_clean "$1" --mode=k8s --keep_hours=6
+}
+
+#######################################
+# Set common variables for the cleanup script.
+#######################################
+cleanup::run_clean() {
+  local job_name="${1:?Usage: cleanup::run_clean job_name}"
```
Comment (XuanWang-Amos) [inline tools/internal_ci/linux/grpc_xds_resource_cleanup.sh:82]: Can we add comment to explain what `job_name` we accept here?

File: tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py [hunk 2]
```diff
@@ -129,7 +170,7 @@ def get_expire_timestamp() -> datetime.datetime:
     )
 
 
-def exec_gcloud(project: str, *cmds: List[str]) -> Json:
+def exec_gcloud(project: str, *cmds: str) -> Json:
```
Comment (XuanWang-Amos) [inline tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py:173]: Should we keep`*cmds` as list?

**Review by XuanWang-Amos — APPROVED** [review]

File: tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py (hunk 2 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py:173]: Nothing changed, it's just the correct way to annotate `*args`. From  https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values:

```py
def foo(*args: str, **kwds: int): ...
```

is acceptable and it means that, e.g., all of the following represent function calls with valid types of arguments:

```py
foo('a', 'b', 'c')
foo(x=1, y=2)
foo('', z=0)
```

---

```py
foo(*cmds: List[str])
```

 would mean that

```py
foo(['a', 'b'], ['c', 'd'], ...)
```

is expected

File: tools/internal_ci/linux/grpc_xds_resource_cleanup.sh (hunk 1 above)
Comment (sergiitk) [reply tools/internal_ci/linux/grpc_xds_resource_cleanup.sh:82]: Ahahah good point. I tried to figure out what it's accepting and realized I completely forgot myself.

**Comment by sergiitk** [comment]
New set of tests:

- [ ] [grpc/core/master/linux/grpc_xds_resource_cleanup](https://source.cloud.google.com/results/invocations/12f34730-501e-49d0-a40e-481684551246)
- [ ] [grpc/java/v1.58.x/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/778196e8-f22c-4ea4-9ea3-c80cbfa8a4c9)
- [ ] [grpc/java/v1.58.x/branch/psm-security](https://source.cloud.google.com/results/invocations/71d9d189-9e84-4496-9f0c-05651fb793fa)
- [ ] [grpc/java/v1.58.x/branch/xds_url_map](https://source.cloud.google.com/results/invocations/bcc20c80-56fb-4eb2-8dc2-0611887afb20)

**Comment by sergiitk** [comment]
Another run after updating resource retention:

- [ ] [grpc/core/master/linux/grpc_xds_resource_cleanup](https://source.cloud.google.com/results/invocations/b32df1ee-5af0-4b7f-8b8e-5227fccaeaa8)

**Pull request merged**